src/test/java/service/impl/OperationHandlerTest.java

```diff
@@ -0,0 +1,80 @@
+package service.impl;
+
+import static org.junit.Assert.assertEquals;
+
+import dao.FruitDao;
+import dao.FruitDaoImpl;
+import java.util.HashMap;
+import java.util.Map;
+import model.FruitTransaction;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import service.OperationHandler;
+import strategy.AddOperationHandler;
+import strategy.SetBalanceOperationHandler;
+import strategy.SubtractOperationHandler;
+
+public class OperationHandlerTest {
+    private static FruitDao fruitDao;
+    private static Map<String, Integer> storage;
+
+    @Rule
+    public ExpectedException exception = ExpectedException.none();
+
+    @BeforeClass
+    public static void initial() {
+        storage = new HashMap<>();
+        fruitDao = new FruitDaoImpl(storage);
+    }
+
+    @Test
+    public void doTransaction_supplyOperationsWorkRight_ok() {
+        String message = "expected quantity and actual are different";
+        OperationHandler operationHandler = new AddOperationHandler(fruitDao);
+        operationHandler.doTransaction(new FruitTransaction("s", "watermelon", 10));
+        assertEquals(message, 10, fruitDao.get("watermelon").intValue());
+    }

```

Comment:
Is better to chek separatly, in separatet test file , for their works with db.
strategy.AddOperationHandler;
strategy.SetBalanceOperationHandler;
strategy.SubtractOperationHandler;

In this test, you test operation handler and returned operation. so you should test only one thing.
You have method doTransaction. invoke it in operation type class